api/src/libs/profile.ts

```diff
@@ -124,70 +125,59 @@ export const updateProfileStatus = async (
   }
 };
 
-export const getQuotaSize = async (
-  profile: ProjectProfile
-): Promise<ProjectQuotaSize> => {
+// Test just to get one namespace quptasize
+export const getNamespaceQuotaSize = async (
+  profile: ProjectProfile,
+  namespace: string
+): Promise<NamespaceQuotaSize> => {
   try {
-    const clusters: Cluster[] = await getClusters(profile);
-
-    const promises: any = [];
-    clusters.forEach((cluster: Cluster) => {
-      if (!profile.id || !cluster.id) {
-        throw new Error("Unable to get profile id or cluster id");
-      }
-
-      promises.push(
-        NamespaceModel.getProjectSetQuotaSize(profile.id, cluster.id)
-      );
-    });
-
-    // Some profile may have two clusters, which has eight namespace
-    const profileQuotaSizes: ProjectQuotaSize[] = await Promise.all(promises);
+    const profileNamespaceQuotaSizesInAllCulster: NamespaceQuotaSize[] =
+      await NamespaceModel.getProfileNamespaceQuotaSize(profile.id, namespace);
 
     let hasSameQuotaSizesForAllClusters: boolean = false;
     // profileQuotaSizes is an array [{ quotaCpuSize: 'small', quotaMemorySize: 'small', quotaStorageSize: 'small', quotaSnapshotSize: 'small' }]
-    if (profileQuotaSizes.length === 1) {
+    if (profileNamespaceQuotaSizesInAllCulster.length === 1) {
       hasSameQuotaSizesForAllClusters = true;
     } else {
-      const quotaSizesForAllClusters: NameSpacesQuotaSize = {
+      const namespaceQuotaSizesForAllClusters: NameSpacesQuotaSize = {
         quotaCpuSize: [],
         quotaMemorySize: [],
         quotaStorageSize: [],
         quotaSnapshotSize: [],
       };
-      const QuotaSizeObjectKey = Object.keys(quotaSizesForAllClusters);
+      const QuotaSizeObjectKey = Object.keys(namespaceQuotaSizesForAllClusters);
       /**
        * following line is to push all quota info from array of object into a single object
        *  that can be consumed by compareNameSpaceQuotaSize to compare if quota size are the same
        * across all cluster.
        */
-      profileQuotaSizes.forEach((element) => {
+      profileNamespaceQuotaSizesInAllCulster.forEach((element) => {
         QuotaSizeObjectKey.forEach((key) => {
-          quotaSizesForAllClusters[key].push(element[key]);
+          namespaceQuotaSizesForAllClusters[key].push(element[key]);
         });
       });
```

Comment:
seems like this can be expressed in a reduce or soemthing. but i'll let it slide for now